research/conditional/utils/conditional_trainer.py

```diff
@@ -121,7 +128,7 @@ def _before_train_operations(self):
             "failure",
         )
 
-    def _after_train_operations(self):
+    def _after_train_operations(self, n_steps: int):
```

Comment:
Can't we access `n_steps` via `self`?

Reply:
Actually I see it's not stored in `self`. Let's put it there and remove the arg from train.

Reply:
It is better to fix this with refactor PR, fixing n_steps via param case is more complicated than scope of this PR.

Reply:
added TODO in code

Reply:
Typically we add stuff like that [here](https://github.com/llm-random/llm-random/issues), but here we can leave it as TODO